src/hotspot/share/prims/jvmtiEnvBase.cpp

```diff
@@ -549,14 +551,19 @@ JvmtiEnvBase::new_jobjectArray(int length, Handle *handles) {
   return objArray;
 }
 
-jthread *
-JvmtiEnvBase::new_jthreadArray(int length, Handle *handles) {
-  return (jthread *) new_jobjectArray(length,handles);
-}
-
 jthreadGroup *
-JvmtiEnvBase::new_jthreadGroupArray(int length, Handle *handles) {
-  return (jthreadGroup *) new_jobjectArray(length,handles);
+JvmtiEnvBase::new_jthreadGroupArray(int length, objArrayHandle groups) {
+  if (length == 0) {
+    return NULL;
```

Comment:
I do not think returning NULL is allowed for JVMTI `GetThreadGroupChildren()`.
Please, see:
[GetThreadGroupChildren](https://docs.oracle.com/en/java/javase/19/docs/specs/jvmti.html#GetThreadGroupChildren)

Reply:
I don't think this has changed. Right now, if there are no child subgroups then *group_count_ptr will be 0 and *groups_ptr will be NULL as there is no memory to deallocate. JVMTI Deallocate is specified to do nothing when called with NULL.

Reply:
Alan, you are right. This check existed before.